src/Simplex/Chat/Util.hs

```diff
@@ -30,3 +33,11 @@ encryptFile fromPath toPath cfArgs = do
 chunkSize :: Num a => a
 chunkSize = 65536
 {-# INLINE chunkSize #-}
+
+shuffle :: [a] -> IO [a]
+shuffle xs = do
+  randomNumbers <- mapM (\_ -> randomRIO (0, 65535) :: IO Word16) xs
+  let zipped = zip randomNumbers xs
+      sorted = sortOn fst zipped
```

Comment:
`sortBy (comparing fst)`?

`sortOn` does extra wrapping/unwrapping for no gain as the keys are numbers already.

Reply:
why not decorate original data while generating randoms?

Reply:
it seems the opposite: https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-List.html#v:sortOn

[sortOn](https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-List.html#v:sortOn) f is equivalent to [sortBy](https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-List.html#v:sortBy) ([comparing](https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-Ord.html#v:comparing) f), but has the performance advantage of only evaluating f once for each element in the input list.

Reply:
I think I understand what you say now - it's better to evaluate fst twice than to move to another tuple... Interesting.